### PR TITLE
feat(ui): add ability to filter search results by DatasetLineageType

### PR DESCRIPTION
**Pull request by vojtechneradatos**
Added ability to filter by DatasetLineageType while searching for a keyword in search bar.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by chriscollins3456** [comment]
hey @vojtechneradatos ! thanks for the PR - we'll be looking at this shortly

File: metadata-models/src/main/pegasus/com/linkedin/dataset/Downstream.pdl [hunk 1]
```diff
@@ -21,5 +21,10 @@ record Downstream {
   /**
    * The type of the lineage
    */
+  @Searchable = {
+    "fieldType": "KEYWORD",
+    "addToFilters": true,
+    "filterNameOverride": "Lineage Type"
+  }
   type: DatasetLineageType
```
Comment (chriscollins3456) [inline metadata-models/src/main/pegasus/com/linkedin/dataset/Downstream.pdl:29]: what would this look like to add as a filter in search?

**Comment by chriscollins3456** [comment]
@vojtechneradatos can you explain your use case for this change a little more for us? Just want to make sure I understand the intention here

**Comment by tomas-hajny** [comment]
> @vojtechneradatos can you explain your use case for this change a little more for us? Just want to make sure I understand the intention here

@chriscollins3456: This is the modification mentioned in [Slack](https://datahubspace.slack.com/archives/CV2UXSE9L/p1672680418271039), I asked my colleague to prepare PRs for our modifications having general use. This minor change has two effects:
1) It allows filtering found datasets based on the lineage type. By default, there are three lineage type values in DataHub (we have added 2 more covering our use cases, but that's not really important here). This change allows filtering datasets having e.g. a COPY lineage documented among them (or TRANSFORMED, etc.).
2) As a rather nice side-effect, it provides possibility to actually see the lineage type in the UI, because this information is not visible in the UI at all as of now. I assume that this second use case may become irrelevant in the future after a possible extension of the UI in this area, but the first one would still apply.

**Review by jjoyce0510 — COMMENTED** [review]
Hi @vojtechneradatos,

We've decided not to accept this contribution (at this time). The product itself is not yet requiring that the edge type of lineage upstreams is searchable. In the future, however, we may reopen this PR if it becomes a requirement for rendering!

I'd recommend that you folks continue to keep these changes locally in a fork, for now.

Thanks for the hard work here - we very much appreciate folks like you helping to drive forward the community. Reach out on Slack if you have any questions!

Cheers
John

**Pull request closed without merging**